lib/parse/compose.ts

```diff
@@ -486,11 +486,11 @@ function validateServiceVolume(
 
 function validateLabels(labels: Dict<string>) {
 	_.keys(labels).forEach((name) => {
-		if (!/^[a-zA-Z0-9.-]+$/.test(name)) {
+		if (!/^[!#-&(-_a-~]+$/.test(name)) {
```

Comment:
> be explicit about what's allowed (instead of preventing what's not allowed)

@dfunckt , I don't see how that can happen here. The character class does specify the allowed characters.

Reply:
You're right, I confused `!` in the character class with `^` -- i.e. I read it as `/^[^#-&(-_a-~]+$/`